internal/legacy/pinhal.go

```diff
@@ -0,0 +1,75 @@
+package legacy
+
+import "errors"
+
+// PinOutput represents a pin hardware abstraction layer for a pin that can output a digital signal.
+// This is an alternative to drivers.PinOutput abstraction which is a function type. Pros and cons
+// of both approaches have been discussed in the [relevant issue]. PinOutput should only be used
+// to expose an initialization function of a driver that receives pins of this type. Ideally
+// driver developers should also expose the initialization with drivers.Pin type:
+//
+//	func New(p1, p2, p3 legacy.PinOutput) *Device {
+//		return NewWithPinfuncs(p1.Set, p2.Set, p3.Set)
+//	}
+//
+//	func NewWithPinfuncs(p1, p2, p3 drivers.PinOutput) *Device {
+//		return &Device{p1:p1, p2:p2, p3:p3}
+//	}
+//
+// [relevant issue]: https://github.com/tinygo-org/drivers/pull/749/files
+type PinOutput interface {
```

Comment:
```suggestion
type PinSeter interface {
```
I have no strong opinion on that, but find it very handy and use it whenever possible, because:
* it seems to be [best practice](https://go.dev/doc/effective_go#interface-names) to name interfaces for one functionality the same like the function, but with adding "-er"
* it is immediately clear to the reader of the code, that this is an interface
* no looking for a better name nor any discussion is needed, it is just a rule
* I personally use this rule also for interfaces with more than one function (if applicable), to earn the benefits, so an alternative is "PinOutputer", if the interface is planned to grow

Reply:
@gen2thomas while I have no opinion on the interface name, I just want to point out this PR puts the interface in `internal/legacy` package, for retirement. So no grow is planned.

Reply:
I've left a general comment on this https://github.com/tinygo-org/drivers/pull/753#issuecomment-3298195326